src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -142,7 +142,9 @@ internal void SetBuffer(byte[] buffer, int offset, int count)
 
         internal bool TryGetBuffer(out ArraySegment<byte> buffer)
         {
-            return _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            bool result = _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            Debug.Assert(result, "Performance: TryGetBuffer failed, which means that reads from the content will copy the data!");
```

Comment:
In  the places you called out, it's not assumed, so there's no assert.

I also wouldn't spend energy right now trying log this. If the goal is to have a set of perf-related logging, there are plenty of other things that are relatively expensive... I don't know why this one particular allocation/copy is being focused on, nor do I know what the action item would be if someone saw it getting logged. Are you planning to run a series of scenarios and monitor it?